### PR TITLE
[operator] support cutom autocert-certPeriod time by days

### PR DESCRIPTION
**Pull request by JaredTan95**
As otel-operator matures, the default 365 days will be at risk when users choose to deploy the operator by autoGenerateCert. Therefore, this PR is intended to provide the ability to configure the expiration time.

I performed local verification in the following way:

1. using custom values to install `operator-chart`, the same with `examples/cutstom-autoGenerateCert-period/values.yaml`
```bash
cd charts/opentelemetry-operator
helm install opentelemetry-operator . -f examples/cutstom-autoGenerateCert-period/values.yaml
```
```yaml
manager:
  collectorImage:
    repository: "otel/opentelemetry-collector-k8s"

admissionWebhooks:
  certManager:
    enabled: false
  autoGenerateCert:
    enabled: true
    # If set to true, new webhook key/certificate is generated on helm upgrade.
    recreate: false
    # Cert period time in years. The default is 1(365 days).
    certPeriodDays: 200
```

3. get `tls.cert`
```bash
kubectl get secret opentelemetry-operator-controller-manager-service-cert -o jsonpath='{.data.tls\.crt}' | base64 --decode > tls.crt
```

4. use `openssl` to check the period time of tls.crt
```bash
openssl x509 -in tls.crt -noout -dates 
```

and I got this:
```txt
notBefore=Jul  9 02:57:00 2024 GMT
notAfter=Jan 25 02:57:00 2025 GMT # 👈 This is in line with my expectations
```

### CODE REVIEW

**Comment by JaredTan95** [comment]
sad, https://github.com/open-telemetry/opentelemetry-helm-charts/actions/runs/9841098833/job/27166919581?pr=1249#step:5:200

Since each CI will re-generated `cutstom-autoGenerateCert-period` example, can not be consistent, remove `cutstom-autoGenerateCert-period`?

File: charts/opentelemetry-operator/values.schema.json [hunk 1]
```diff
@@ -1571,11 +1571,20 @@
                             "examples": [
                                 true
                             ]
+                        },
+                        "certPeriodYear": {
```
Comment (jaronoff97) [inline charts/opentelemetry-operator/values.schema.json:1575]: why are we using years here if we're just going to convert this to days? IMO we should only use days

File: charts/opentelemetry-operator/examples/cutstom-autoGenerateCert-period/values.yaml
Comment (jaronoff97) [inline charts/opentelemetry-operator/examples/cutstom-autoGenerateCert-period/values.yaml:1]: the name of this example should be `custom-auto-generate-cert-period` IMO

Comment (jaronoff97) [reply charts/opentelemetry-operator/examples/cutstom-autoGenerateCert-period/values.yaml:1]: also, do we need to have a full example for setting this single variable?

Comment (JaredTan95) [reply charts/opentelemetry-operator/examples/cutstom-autoGenerateCert-period/values.yaml:1]: I think `examples/cutstom-autoGenerateCert-period/values.yaml ` content is the full example

File: charts/opentelemetry-operator/values.schema.json (hunk 1 above)
Comment (jaronoff97) [reply charts/opentelemetry-operator/values.schema.json:1575]: IMO we should stick with days as its more granular, ex: i have expirations set to 290 days for some of my company.

File: charts/opentelemetry-operator/examples/cutstom-autoGenerateCert-period/values.yaml
Comment (jaronoff97) [reply charts/opentelemetry-operator/examples/cutstom-autoGenerateCert-period/values.yaml:1]: yes, i'm not sure we need to make a whole new full example for this, i was mentioning that the naming with caps is a bit odd here too

File: charts/opentelemetry-operator/values.schema.json (hunk 1 above)
Comment (JaredTan95) [reply charts/opentelemetry-operator/values.schema.json:1575]: oh, day more flexible, I will update it.

**Comment by JaredTan95** [comment]
> sad, https://github.com/open-telemetry/opentelemetry-helm-charts/actions/runs/9841098833/job/27166919581?pr=1249#step:5:200
> 
> Since each CI will re-generated `cutstom-autoGenerateCert-period` example, can not be consistent, remove `cutstom-autoGenerateCert-period`?

@jaronoff97 hi, any idea about this?

**Comment by jaronoff97** [comment]
@JaredTan95 what do you get if you run the make generate locally?

**Comment by JaredTan95** [comment]
every time run ``

> @JaredTan95 what do you get if you run the make generate locally?

Similarly, the certificate is regenerated each time with `CHARTS=opentelemetry-operator make generate-examples`

**Comment by jaronoff97** [comment]
@JaredTan95 seems like we have failures [here ](https://github.com/open-telemetry/opentelemetry-helm-charts/pull/1251)too same thing.

**Comment by jaronoff97** [comment]
ohhh i see the problem now... hmmm.....

**Comment by TylerHelmuth** [comment]
Let's not add the new problematic example. The scenario is pretty straight forward (as in the values.yaml file isn't common or interesting) so I don't think we need a new example for it.

**Review by TylerHelmuth — APPROVED** [review]

**Pull request merged**